prost-build/src/code_generator.rs

```diff
@@ -546,47 +587,40 @@ impl<'a> CodeGenerator<'a> {
         &mut self,
         message_name: &str,
         fq_message_name: &str,
-        oneof: &OneofDescriptorProto,
-        fields: &[(FieldDescriptorProto, usize)],
+        oneof: &OneofField,
     ) {
-        let name = format!(
+        let type_name = format!(
             "{}::{}",
             to_snake(message_name),
-            to_upper_camel(oneof.name())
+            to_upper_camel(oneof.descriptor.name())
         );
+        let field_tags = oneof
```

Comment:
I don't like this separated variable. I prefer the old style where the iter/join/map combination is located where it is used.